userspace/ksud/src/utils.rs

```diff
@@ -1,393 +1,400 @@
-        let relative_path = source_path
-            .strip_prefix(source.as_ref())
-            .context("Failed to generate relative path")?;
-        let dest_path = destination.as_ref().join(relative_path);
-
-        if let Some(parent) = dest_path.parent() {
-            std::fs::create_dir_all(parent).context("Failed to create directory")?;
-        }
-
-        if entry.file_type().is_file() {
-            std::fs::copy(&source_path, &dest_path).with_context(|| {
-                format!("Failed to copy file from {source_path:?} to {dest_path:?}",)
-            })?;
-            copy_xattrs(&source_path, &dest_path)?;
-        } else if entry.file_type().is_symlink() {
-            if dest_path.exists() {
-                std::fs::remove_file(&dest_path).context("Failed to remove file")?;
-            }
-            let target = std::fs::read_link(entry.path()).context("Failed to read symlink")?;
-            log::info!("Symlink: {:?} -> {:?}", dest_path, target);
-            std::os::unix::fs::symlink(target, &dest_path).context("Failed to create symlink")?;
-            copy_xattrs(&source_path, &dest_path)?;
-        } else if entry.file_type().is_dir() {
-            create_dir_all(&dest_path)?;
-            let metadata = std::fs::metadata(&source_path).context("Failed to read metadata")?;
-            std::fs::set_permissions(&dest_path, metadata.permissions())
-                .with_context(|| format!("Failed to set permissions for {dest_path:?}"))?;
-            copy_xattrs(&source_path, &dest_path)?;
-        } else if entry.file_type().is_char_device() {
-            if dest_path.exists() {
-                std::fs::remove_file(&dest_path).context("Failed to remove file")?;
-            }
-            let metadata = std::fs::metadata(&source_path).context("Failed to read metadata")?;
-            let mode = metadata.permissions().mode();
-            let dev = metadata.rdev();
-            if dev == 0 {
-                log::info!(
-                    "Found a char device with major 0: {}",
-                    entry.path().display()
-                );
-                rustix::fs::mknodat(
-                    rustix::fs::CWD,
-                    &dest_path,
-                    rustix::fs::FileType::CharacterDevice,
-                    mode.into(),
-                    dev,
-                )
-                .with_context(|| format!("Failed to create device file at {dest_path:?}"))?;
-                copy_xattrs(&source_path, &dest_path)?;
-            }
-        } else {
-            log::info!(
-                "Unknown file type: {:?}, {:?},",
-                entry.file_type(),
-                entry.path(),
-            );
-        }
-    }
-    Ok(())
-}
-
-#[cfg(not(any(target_os = "linux", target_os = "android")))]
-pub fn copy_module_files(_source: impl AsRef<Path>, _destination: impl AsRef<Path>) -> Result<()> {
-    unimplemented!()
-}
+use anyhow::{bail, Context, Error, Ok, Result};

```

Comment:
the format seems incorrect, can you run `cargo format` first?

Reply:
I have not made any changes to the copy_module_files function, which comes from the original userspace/ksud/src/utils.rs file.  The whole file is marked red seems to be a bug

Reply:
In order to solve the difference comparison problem, I reopened a new PR, please refer to the new PR.